src/farming/library.cairo

```diff
@@ -0,0 +1,1039 @@
+    }() -> (carbonable_project_address: felt) {
+        let (carbonable_project_address) = CarbonableFarming_carbonable_project_address_.read();
+        return (carbonable_project_address=carbonable_project_address);
+    }
+
+    func carbonable_project_slot{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        ) -> (carbonable_project_slot: Uint256) {
+        let (carbonable_project_slot) = CarbonableFarming_carbonable_project_slot_.read();
+        return (carbonable_project_slot=carbonable_project_slot);
+    }
+
+    func total_deposited{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+        value: Uint256
+    ) {
+        let (value) = CarbonableFarming_total_registered_value_.read();
+        return (value=value);
+    }
+
+    func total_absorption{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+        absorption: felt
+    ) {
+        let (computed) = _compute_total_absorption();
+        let (stored) = CarbonableFarming_total_absorption_.read();
+        return (absorption=computed + stored);
+    }
+
+    func max_absorption{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+        absorption: felt
+    ) {
+        // [Compute] Max absorption, which is the whole project absorption until now
+        let (contract_address) = CarbonableFarming_carbonable_project_address_.read();
+        let (slot) = CarbonableFarming_carbonable_project_slot_.read();
+        let (absorption) = ICarbonableProject.getCurrentAbsorption(
+            contract_address=contract_address, slot=slot
+        );
+        return (absorption=absorption);
+    }
+
+    func total_sale{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+        sale: felt
+    ) {
+        alloc_locals;
+
+        // [Check] Price is setup, otherwise return 0
+        let (len) = Array.read_len(key=TIME_SK);
+        if (len == 0) {
+            return (sale=0);
+        }
+
+        // [Check] Ton equivalent is not zero
+        let (contract_address) = CarbonableFarming_carbonable_project_address_.read();
+        let (slot) = CarbonableFarming_carbonable_project_slot_.read();
+        let (ton_equivalent) = ICarbonableProject.getTonEquivalent(
+            contract_address=contract_address, slot=slot
+        );
+        if (ton_equivalent == 0) {
+            return (sale=0);
+        }
+
+        // [Compute] Total sale
+        let (computed) = _compute_total_sale();
+        let (stored) = CarbonableFarming_total_sale_.read();
+        let (sale_in_dollars, _) = unsigned_div_rem(computed + stored, ton_equivalent);
+
+        return (sale=sale_in_dollars);
+    }
+
+    func max_sale{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+        sale: felt
+    ) {
+        alloc_locals;
+
+        // [Check] Price is setup, otherwise return 0
+        let (len) = Array.read_len(key=TIME_SK);
```

Comment:
If I follow the comment, the check should be key=PRICE_SK instead of key=TIME_SK ?

Reply:
Same have above.